src/model/test/CoilUserDefined_GTest.cpp

```diff
@@ -0,0 +1,246 @@
+#include "../CoilUserDefined_Impl.hpp"
+#include "../EnergyManagementSystemProgramCallingManager.hpp"
+#include "../EnergyManagementSystemProgram.hpp"
+#include "../EnergyManagementSystemActuator.hpp"
+#include "../ThermalZone.hpp"
+#include "../Node.hpp"
+#include "../Node_Impl.hpp"
+
+#include "../../utilities/core/Path.hpp"
+#include "../../utilities/idf/IdfFile.hpp"
+#include "../../utilities/idf/IdfObject.hpp"
+
+//using namespace openstudio::energyplus;
+using namespace openstudio;
+using namespace openstudio::model;
+
+TEST_F(ModelFixture, CoilUserDefined_CoilUserDefined) {
+  ::testing::FLAGS_gtest_death_test_style = "threadsafe";
+
+  ASSERT_EXIT(
+    {
+      Model m;
+      CoilUserDefined coil_ud(m);
+
+      exit(0);
+    },
+    ::testing::ExitedWithCode(0), "");
+}
+
+TEST_F(ModelFixture, CoilUserDefined_constructor) {
+  Model m;
+
+  CoilUserDefined b1(m);
+
+  EXPECT_TRUE(b1.airOutletTemperatureActuator());
+  EXPECT_TRUE(b1.airOutletHumidityRatioActuator());
+  EXPECT_TRUE(b1.airMassFlowRateActuator());
+  EXPECT_TRUE(b1.plantMinimumMassFlowRateActuator());
+  EXPECT_TRUE(b1.plantMaximumMassFlowRateActuator());
+  EXPECT_TRUE(b1.plantDesignVolumeFlowRateActuator());
+  EXPECT_TRUE(b1.plantMassFlowRateActuator());
+  EXPECT_TRUE(b1.plantOutletTemperatureActuator());
+
+  EXPECT_TRUE(b1.airOutletTemperatureActuator().get().actuatedComponent());
+  EXPECT_EQ(b1.handle(), b1.airOutletTemperatureActuator().get().actuatedComponent().get().handle());
+}
+
+TEST_F(ModelFixture, CoilUserDefined_programs) {
+  Model m;
+
+  //PlantLoop plant(m);
+  CoilUserDefined b1(m);
+  EnergyManagementSystemProgramCallingManager mainPCM(m);
+  EnergyManagementSystemProgramCallingManager initPCM(m);
+  EXPECT_TRUE(b1.setOverallModelSimulationProgramCallingManager(mainPCM));
+  EXPECT_EQ(mainPCM, b1.overallModelSimulationProgramCallingManager().get());
+  EXPECT_TRUE(b1.setModelSetupandSizingProgramCallingManager(initPCM));
+  EXPECT_EQ(initPCM, b1.modelSetupandSizingProgramCallingManager().get());
+  b1.resetOverallModelSimulationProgramCallingManager();
+  b1.resetModelSetupandSizingProgramCallingManager();
+  EXPECT_FALSE(b1.overallModelSimulationProgramCallingManager());
+  EXPECT_FALSE(b1.modelSetupandSizingProgramCallingManager());
+}
+
+TEST_F(ModelFixture, CoilUserDefined_addToNode) {
+  Model m;
+
+  CoilUserDefined coil(m);
+
+  model::AirLoopHVAC airLoop(m);
+
+  model::Node supplyOutletNode = airLoop.supplyOutletNode();
+
+  coil.addToNode(supplyOutletNode);
+
+  ASSERT_EQ((unsigned)3, airLoop.supplyComponents().size());
+}
+
+// Add CoilUserDefined to AirLoopHVAC and PlantLoop
+// Test CoilUserDefined::remove and make sure that the loops are intact
+TEST_F(ModelFixture, CoilUserDefined_remove) {
+  Model m;
+  CoilUserDefined coil(m);
+
+  AirLoopHVAC airLoop(m);
+  Node supplyOutletNode = airLoop.supplyOutletNode();
+
+  coil.addToNode(supplyOutletNode);
+  coil.remove();
+  ASSERT_EQ((unsigned)2, airLoop.supplyComponents().size());
```

Comment:
you can just use `2` in there. Gtest accepts it without flinching now (it used not to). And C-style casts are terrible, so if really wanted to specify it's unsigned, use `2u`